src/main/java/org/truffleruby/core/array/DoubleArrayMirror.java

```diff
@@ -65,6 +67,10 @@ public ArrayMirror extractRange(int start, int end) {
         return new DoubleArrayMirror(ArrayUtils.extractRange(array, start, end));
     }
 
+    public void sort(int size) {
```

Comment:
There should be a TruffleBoundary on these Arrays.sort calls. I wonder why it doesn't fail in CI as it should increase the number of methods significantly or even recurse infinitely. Could you check the primitive sorting specialization is indeed taken for e.g. int[] ?

Reply:
I can confirm the specialisation was being taken for `int[]`.